### PR TITLE
[Chore] Add go fmt check to linter

### PR DESCRIPTION
**Pull request by brandonc**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,13 +6,11 @@ generate: check-resource
...
 	go run . $(RESOURCE) ;
 
-fmt:
-
 vet:
 	go vet
 
 fmt:
-	go fmt ./...
+	gofmt -s -l -w .
```
Comment (sebasslash) [inline Makefile:13]: For context:
`-s` tells `gofmt` to simplify your code
`-l` tells `gofmt` to list the files that changed due to formatting
`-w` tells `gofmt` to write the formatting changes to the source file

Run `gofmt -help` for more info!

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ There are instances where several new resources being added (i.e Workspace Run T
 
 ## Linting
 
-Linting is not necessarily required for a change to be merged, but it helps smooth the review process and catch common mistakes early. If you'd like to run the linters manually, follow these steps:
+Linting is not necessarily required for a change to be merged, but it helps smooth the review process and catch common mistakes early. One important linter checks that `go fmt` has been run. Use `make fmt` to automatically format your source files before committing. If you'd like to run the linters manually, follow these steps:
```
Comment (sebasslash) [inline docs/CONTRIBUTING.md:21]: Non blocker: Maybe change the first sentence to
`A workflow will execute upon opening a pull request to ensure `go fmt` has been run.`

**Review by sebasslash — COMMENTED** [review]
I think it looks good to me :heavy_check_mark: 

I have a suggestion on the wording in our docs :arrow_down:

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (brandonc) [reply docs/CONTRIBUTING.md:21]: Yeah I see your point I forgot to mention it

**Review by sebasslash — APPROVED** [review]
Awesome, like this wording much better :fire:

**Review by annawinkler — APPROVED** [review]
:shipit:

**Pull request merged**